File: go/module-config/pull-with-push-override/fs/foo-sfl.go
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the Apache 2.0 license
 *  that can be found in the LICENSE file.
 */

package fs

import (
	"fmt"
	"github.com/pvillela/moduleconfig/go/module-config/pull-with-push-override/config"
)

type FooSflCfgInfo struct {
	X string
}

type FooSflCfgSrc = config.CfgSrc[FooSflCfgInfo]

var FooSflCfgSrcV = config.MakeCfgSrc[FooSflCfgInfo](FooSflCfgAdapter)

func FooSfl() {
	fmt.Println(FooSflCfgSrcV().X)
	BarBf()
}

File: ts/module-config/pull-with-push-override/fs/foo-sfl.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

import { barBf } from "./bar-bf";
import { identity, makeCfgSrc } from "../config/cfg-src";

type FooSflCfgInfo = {
  x: string;
};

export const fooSflCfgSrc = makeCfgSrc<FooSflCfgInfo>(identity)

export function fooSfl() {
  console.log(fooSflCfgSrc().x);
  barBf();
}
